rs/messaging/src/routing/stream_handler.rs

```diff
@@ -786,6 +786,12 @@ impl StreamHandlerImpl {
                     unreachable!("No signals are generated for response induction failures except for CanisterMigrating");
                 }
             }
+        } else if matches!(msg, RequestOrResponse::Request(_)) {
+            // `remote_subnet_id` is not known to be a valid host for `msg.sender()`.
+            // This can legitimatly happen if the sender canister is migrating.
+            self.observe_inducted_message_status(msg_type, LABEL_VALUE_SENDER_SUBNET_MISMATCH);
```

Comment:
Same as in the other PR, I'd use a new label here to differentiate between the case below (definite bug or malicious sender; message silently dropped) and this one (as far as we know, a canister migration; request rejected).